### PR TITLE
 LubimovaYE lab1

### PR DESCRIPTION
**Pull request by koreshh**

### CODE REVIEW

File: LubimovaYE/lab_1/main.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+int main()
+{
+	
+	float operand_1, operand_2;
+	char operation_sign;
+	cout << "Enter 2 numbers and an action sign(+,-,*,/)" << endl;
+	cout << "To exit the program, enter '-1' three times" << endl;
+	cin >> operand_1 >> operand_2 >> operation_sign;
+	while (operand_1 != -1 && operand_2 != -1 && operation_sign != '-1')
```
Comment (Toxa-man) [inline LubimovaYE/lab_1/main.cpp:12]: If I try to calculate sum of -1 and 2  your program exits which is not an expected behavior.
image

File: LubimovaYE/lab_1/main.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	{
+		switch (operation_sign)
+		{
+		case '+':
+			cout << operand_1 + operand_2 << endl;
```
Comment (Toxa-man) [inline LubimovaYE/lab_1/main.cpp:17]: Please extract `cout << operand_1 + operand_2 << endl;` and `cout << "Enter new values" << endl;` from switch..case in order to reduce code duplication

**Review by Toxa-man — COMMENTED** [review]
Code issues

File: LubimovaYE/lab_1/main.cpp (hunk 2 above)
Comment (koreshh) [reply LubimovaYE/lab_1/main.cpp:17]: #include <iostream>
using namespace std;

int main()
{
	
	float operand_1, operand_2, res;
	bool flag;
	flag = 0;
	char operation_sign;
	cout << "Enter 2 numbers and an action sign(+,-,*,/)" << endl;
	cout << "To exit the program, enter '0' three times" << endl;
	cin >> operand_1 >> operand_2 >> operation_sign;
	while (!(operand_1 == 0 && operand_2 == 0 && operation_sign == '0'))
	{
		switch (operation_sign)
		{
		case '+':
			res = operand_1 + operand_2;
			break;
		case '-':
			res = operand_1 - operand_2;
			break;
		case '*':
			res = operand_1 * operand_2;
			break;
		case '/':
			if (operand_2 == 0)
			{
				flag = 1;
				cout << "You can't divide by 0" << endl;
			}
			else
			{
				res = operand_1 / operand_2;
			}
			break;
		default:
			cout << "Enter a valid operation sign" << endl;
			cout << "Enter new values" << endl;
			break;
		}
		if (flag!=1)
		{
			cout << res << endl;
		}
		res = 0;
		flag = 0;
		cout << "Enter new values" << endl;
		cin >> operand_1 >> operand_2 >> operation_sign;
	}
	cout << "The programme is completed" << endl;
	return 0;
}

**Pull request closed without merging**